packages/terra-table/CHANGELOG.md

```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Changed
+  * Re-implementing Terra Table in preparation of being consumed by DataGrid package.
+
```

Comment:
A few things here:

1. Since we are releasing this component for consumers, we should use consumer-oriented language. Yes, the motivation is to be consumed by DataGrid for _now_ but we still releasing this for consumers so we should describe the changes from a consumer's perspective
2. These are breaking changes so we should put them under the `Breaking Changes` header.

Reply:
Done!